contracts/nft/src/execute/mint.rs

```diff
@@ -0,0 +1,167 @@
+use std::str::FromStr;
+
+use cosmwasm_std::{Addr, DepsMut, Env, MessageInfo, Response};
+use provwasm_std::types::provenance::metadata::v1::process::ProcessId;
+use provwasm_std::types::provenance::metadata::v1::record_input::Source;
+use provwasm_std::types::provenance::metadata::v1::{
+    input_specification, DefinitionType, InputSpecification, MsgWriteRecordRequest,
+    MsgWriteRecordSpecificationRequest, MsgWriteScopeRequest, MsgWriteSessionRequest, Party,
+    PartyType, Process, Record, RecordInput, RecordInputStatus, RecordOutput, RecordSpecification,
+    ResultStatus, Scope, Session,
+};
+use sha2::{Digest, Sha256};
+use uuid::Uuid;
+
+use crate::storage::nft::{Nft, TOKENS};
+use crate::storage::nft_count;
+use crate::util::metadata_address::MetadataAddress;
+use crate::{
+    core::error::ContractError,
+    storage,
+    util::action::{Action, ActionType},
+};
+
+pub fn handle(
+    deps: DepsMut,
+    _info: MessageInfo,
+    env: Env,
+    scope_uuid: Uuid,
+    session_uuid: Uuid,
+    recipient: Addr,
+) -> Result<Response, ContractError> {
+    let state = storage::state::get(deps.storage)?;
```

Comment:
Does any sort of check need to be done for who mints? I noticed you check for permissions for send.

Reply:
should be fixed now